internal/test/framework/framework.go

```diff
@@ -73,91 +79,43 @@ type TestFramework struct {
 	// clusterAddress is the address of the OpenShift cluster e.g. "foo.fah.com".
 	// This should not include "https://api-" or a port.
 	ClusterAddress string
-}
-
-// Creds is used for parsing the vmCreds command line argument
-type Creds []*types.Credentials
-
-// Set populates the list of credentials from the vmCreds command line argument
-func (c *Creds) Set(value string) error {
-	if value == "" {
-		return nil
-	}
-
-	splitValue := strings.Split(value, ",")
-	// Credentials consists of three elements, so this has to be
-	if len(splitValue)%3 != 0 {
-		return fmt.Errorf("incomplete VM credentials provided")
-	}
-
-	// TODO: Add input validation if we want to use this in production
-	// TODO: Change username based on cloud provider if this is to be used for clouds other than AWS
-	for i := 0; i < len(splitValue); i += 3 {
-		cred := types.NewCredentials(splitValue[i], splitValue[i+1], splitValue[i+2], awsUsername)
-		*c = append(*c, cred)
-	}
-	return nil
-}
-
-// String returns the string representation of Creds. This is required for Creds to be used with flags.
-func (c *Creds) String() string {
-	return fmt.Sprintf("%v", *c)
+	// Signer is a signer created from the user's private key
+	Signer ssh.Signer
+	// client for interacting with Kubernetes API servers
+	client client.Client
+	// machineSet holds the MachineSet configuration used to destroy MachineSets
+	machineSet *mapi.MachineSet
 }
 
 // initCIvars gathers the values of the environment variables which configure the test suite
 func initCIvars() error {
 	kubeconfig = os.Getenv("KUBECONFIG")
-	if kubeconfig == "" {
-		return fmt.Errorf("KUBECONFIG environment variable not set")
-	}
-	awsCredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
-	if awsCredentials == "" {
-		return fmt.Errorf("AWS_SHARED_CREDENTIALS_FILE environment variable not set")
-	}
 	artifactDir = os.Getenv("ARTIFACT_DIR")
-	if awsCredentials == "" {
+	if artifactDir == "" {
```

Comment:
Is this check required, as we create. a temp ARTIFACT_DIR if the env variable is not set.